include/cynq/memory.hpp

```diff
@@ -47,6 +48,20 @@ class IMemory {
    * @return Status
    */
   virtual Status Sync(const SyncType type) = 0;
+  /**
+   * @brief Sync method (Asynchronous)
```

Comment:
for now this is fine, maybe adding await instead of using the word async for this might be better as a naming convention?

Reply:
I disagree with the pythonic / javascript spelling. For hardware acceleration, usually we use async and sync.

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html